pkg/client/add_events.go

```diff
@@ -250,22 +251,30 @@ func (client *DataSetClient) Shutdown() error {
 		lastProcessed = client.eventsProcessed.Load()
 		backoffDelay := expBackoff.NextBackOff()
 		client.Logger.Info(
-			"Not all events has been processed",
+			"Shutting down - processing events",
 			zap.Int("retryNum", retryNum),
 			zap.Duration("backoffDelay", backoffDelay),
 			zap.Uint64("eventsEnqueued", client.eventsEnqueued.Load()),
 			zap.Uint64("eventsProcessed", client.eventsProcessed.Load()),
 		)
 		if backoffDelay == expBackoff.Stop {
-			lastError = fmt.Errorf("not all events has been processed")
+			lastError = fmt.Errorf("not all events have been processed")
+			client.Logger.Error(
```

Comment:
Thanks and yeah makes more sense and it's more useful to log actual counts here :)